src/conv-core/cpuaffinity.C

```diff
@@ -204,8 +204,12 @@ int CmiSetCPUAffinity(int mycore)
   cmi_hwloc_topology_init(&topology);
   cmi_hwloc_topology_load(topology);
 
-  hwloc_cpuset_t cpuset = cmi_hwloc_bitmap_alloc();
-  cmi_hwloc_bitmap_set(cpuset, core);
+  int thread_unitcount = cmi_hwloc_get_nbobjs_by_type(topology, HWLOC_OBJ_PU);
+  int thread_assignment = core % thread_unitcount;
+
+  hwloc_obj_t thread_obj = cmi_hwloc_get_obj_by_type(topology, HWLOC_OBJ_PU, thread_assignment);
+  hwloc_cpuset_t cpuset = cmi_hwloc_bitmap_dup(thread_obj->cpuset);
+  cmi_hwloc_bitmap_singlify(cpuset);
```

Comment:
Can you add some comments to this saying what the hwloc functions are each doing? I, for example, am unclear on why singlify needs to be called at all. Shouldn't there just be one bit set in the first place?